File: languages/csharp_unity_json_properties/Narrative.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Narrative : Element
{
    [JsonProperty("story"), TextAttribute("")]
    public string story;
    [JsonProperty("consequences"), TextAttribute("")]
    public string consequences;
    [JsonProperty("start_date"), Integer(0)]
    public int startDate;
    [JsonProperty("end_date"), Integer(0)]
    public int endDate;
    [JsonProperty("order"), Integer(0)]
    public int order;
    [JsonProperty("parent_narrative"), ReferenceAttribute(typeof(Narrative))]
    public string parentNarrative;
    [JsonProperty("protagonist"), ReferenceAttribute(typeof(Character))]
    public string protagonist;
    [JsonProperty("antagonist"), ReferenceAttribute(typeof(Character))]
    public string antagonist;
    [JsonProperty("narrator"), ReferenceAttribute(typeof(Character))]
    public string narrator;
    [JsonProperty("conservator"), ReferenceAttribute(typeof(Institution))]
    public string conservator;
    [JsonProperty("events"), ReferenceAttribute(typeof(Event), true)]
    public string events;
    [JsonProperty("characters"), ReferenceAttribute(typeof(Character), true)]
    public string characters;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("institutions"), ReferenceAttribute(typeof(Institution), true)]
    public string institutions;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("collectives"), ReferenceAttribute(typeof(Collective), true)]
    public string collectives;
    [JsonProperty("territories"), ReferenceAttribute(typeof(Territory), true)]
    public string territories;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("families"), ReferenceAttribute(typeof(Family), true)]
    public string families;
    [JsonProperty("relations"), ReferenceAttribute(typeof(Relation), true)]
    public string relations;
    [JsonProperty("titles"), ReferenceAttribute(typeof(Title), true)]
    public string titles;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
    [JsonProperty("laws"), ReferenceAttribute(typeof(Law), true)]
    public string laws;
}

File: languages/django_models/narrative.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Narrative(AbstractElementModel):

    # Context
    story = models.TextField(blank=True, null=True)
    consequences = models.TextField(blank=True, null=True)
    start_date = models.PositiveIntegerField(blank=True, null=True)
    end_date = models.PositiveIntegerField(blank=True, null=True)
    order = models.PositiveIntegerField(blank=True, null=True)
    parent_narrative = models.ForeignKey("Narrative", on_delete=models.SET_NULL, blank=True, null=True, related_name="narrative_parent_narrative")
    protagonist = models.ForeignKey("Character", on_delete=models.SET_NULL, blank=True, null=True, related_name="narrative_protagonist")
    antagonist = models.ForeignKey("Character", on_delete=models.SET_NULL, blank=True, null=True, related_name="narrative_antagonist")
    narrator = models.ForeignKey("Character", on_delete=models.SET_NULL, blank=True, null=True, related_name="narrative_narrator")
    conservator = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="narrative_conservator")

    # Involves
    events = models.ManyToManyField("Event", blank=True, related_name="narrative_events")
    characters = models.ManyToManyField("Character", blank=True, related_name="narrative_characters")
    objects = models.ManyToManyField("Object", blank=True, related_name="narrative_objects")
    locations = models.ManyToManyField("Location", blank=True, related_name="narrative_locations")
    species = models.ManyToManyField("Species", blank=True, related_name="narrative_species")
    creatures = models.ManyToManyField("Creature", blank=True, related_name="narrative_creatures")
    institutions = models.ManyToManyField("Institution", blank=True, related_name="narrative_institutions")
    traits = models.ManyToManyField("Trait", blank=True, related_name="narrative_traits")
    collectives = models.ManyToManyField("Collective", blank=True, related_name="narrative_collectives")
    zones = models.ManyToManyField("Zone", blank=True, related_name="narrative_zones")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="narrative_abilities")
    phenomena = models.ManyToManyField("Phenomenon", blank=True, related_name="narrative_phenomena")
    languages = models.ManyToManyField("Language", blank=True, related_name="narrative_languages")
    families = models.ManyToManyField("Family", blank=True, related_name="narrative_families")
    relations = models.ManyToManyField("Relation", blank=True, related_name="narrative_relations")
    titles = models.ManyToManyField("Title", blank=True, related_name="narrative_titles")
    constructs = models.ManyToManyField("Construct", blank=True, related_name="narrative_constructs")
    laws = models.ManyToManyField("Law", blank=True, related_name="narrative_laws")

    def __str__(self):
        return self.name
